variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/ws/VariantSourceEntryWithSampleNames.java

```diff
@@ -61,6 +64,21 @@ public VariantSourceEntryWithSampleNames(String fileId, String studyId, String[]
         }
     }
 
+
+    public VariantSourceEntryWithSampleNames(IVariantSourceEntry variantSourceEntry) {
+        super(variantSourceEntry.getFileId(), variantSourceEntry.getStudyId(),
+              variantSourceEntry.getSecondaryAlternates(), variantSourceEntry.getFormat(),
+              variantSourceEntry.getCohortStats(), variantSourceEntry.getAttributes());
+
+        Set<String> sampleNames = new HashSet<>();
+        for (Map<String, String> stringStringMap : variantSourceEntry.getSamplesData()) {
```

Comment:
getSamplesData() returns a List<Map<String, String>>. Each element of the list, is a Map<String, String> then, that represent the attributes of a given sample:

- map keys are attribute names (like GT, DP...)
- map values are the values for each attribute in the sample

That is, all the columns >=9 in a VCF, where column 9 (FORMAT) has the attribute names and each of the other columns has the values for those attributes for a given sample.

This code is making the wrong assumption that the keys in the Map are the sample Names (see line 75). We need the sample names in a separate List. If we have this sample names list, we can use the VariantWiithSourceEntryWithSampleNames(IVariantSourceEntry, List<String>) constructor and this one is not needed then.